packages/financial-templates-lib/src/logger/PagerDutyTransportV2.ts

```diff
@@ -0,0 +1,49 @@
+// This transport enables winston logging to send messages to pager duty v2 api.
+import Transport from "winston-transport";
+import { event } from "@pagerduty/pdjs";
+
+type TransportOptions = ConstructorParameters<typeof Transport>[0];
+export type Severity = "critical" | "error" | "warning" | "info";
+
+interface Config {
+  integrationKey: string;
+  customServices?: { [key: string]: string };
+}
+
+export class PagerDutyTransportV2 extends Transport {
+  private readonly integrationKey: string;
+  private readonly customServices: { [key: string]: string };
+  constructor(winstonOpts: TransportOptions, { integrationKey, customServices = {} }: Config) {
+    super(winstonOpts);
+    this.integrationKey = integrationKey;
+    this.customServices = customServices;
+  }
+  // pd v2 severity only supports critical, error, warning or info.
+  public static convertLevelToSeverity(level?: string): Severity {
+    if (!level) return "error";
+    if (level === "warn") return "warning";
+    if (level === "info" || level === "critical") return level;
+    return "error";
+  }
+  // Note: info must be any because that's what the base class uses.
+  async log(info: any, callback: () => void): Promise<void> {
+    try {
+      await event({
+        data: {
+          routing_key: this.integrationKey,
```

Comment:
The routing seems to be a bit different, can you explain how this works?

In the previous version, we had a pagerduty client that we would create _with_ the credentials. Then we would provide the name of the service for each event that we wanted to emit. In this version, it appears as if we're not using a client at all and just providing a single key for the route. Does this mean that customServices should be a mapping from a configured route name to an integration key that will send alerts to a particular PD service?

Reply:
ye I asked a similar question. if this is the case then we need some switching logic on `notificationPath`

Reply:
So from what im understanding, the v2 api is more of a logging system, not an alerting system. i think the point is to get all the data into their logs, and then make routing rules based on those which can then set up alerting.  The "routing key" in this case, theres only a single key for your pd account, and i dont see any option to create others. 

As for custom services, i dont know what that does really, but i figured if i passed though as much raw data as i could in the logs, i could create the rules in pd to do proper sorting and routing.